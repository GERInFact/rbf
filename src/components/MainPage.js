import React, { Fragment } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import ErrorBoudry from "../components/ErrorBoundry";

import "./mainPage.css";



class MainPage extends React.Component {
  render() {
    const filteredRobots = this.getFilteredRobots();
    const { onSearchChange, isPending } = this.props;

    return (
      <Fragment>
        <header className="header">
          <h1 className="header_title">RBF</h1>
          <SearchBox searchChange={onSearchChange} />
        </header>
        {!isPending ? (
          <main className="main">
            <ErrorBoudry>
              <CardList users={filteredRobots} />
            </ErrorBoudry>
          </main>
        ) : (
          <h2>Loading...</h2>
        )}
      </Fragment>
    );
  }

  getFilteredRobots() {
    const { searchField, robots } = this.props;
    return robots.filter((r) =>
      r.name.toLowerCase().includes(searchField.toLowerCase())
    );
  }

  componentDidMount() {
    this.props.onRequestRobots();
  }
}

//connect returns another function -> higher order function
export default MainPage;
