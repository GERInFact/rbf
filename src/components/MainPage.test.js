import React from "react";
import renderer from "react-test-renderer";
import MainPage from "./MainPage";


let component;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: "",
        isPending: false
    };
    component = renderer.create(<MainPage {...mockProps}/>);
});

it("renders MainPage without crashing", () => {
    expect(component.toJSON()).toMatchSnapshot();
});

it("filters robots correct", () => {
   expect(component.getInstance().getFilteredRobots()).toEqual([]); 
});