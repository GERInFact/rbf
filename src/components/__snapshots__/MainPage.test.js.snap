// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
Array [
  <header
    className="header"
  >
    <h1
      className="header_title"
    >
      RBF
    </h1>
    <div
      className="search-box"
    >
      <input
        aria-label="Search field for robots"
        className="search-box_input"
        placeholder="search robots"
        type="search"
      />
    </div>
  </header>,
  <main
    className="main"
  >
    <h2>
      Nothing found!
    </h2>
  </main>,
]
`;
